server/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -1258,7 +1262,7 @@ U createIndex(IndexMetadata indexMetadata, List<IndexEventListener> builtInIndex
          * Deletes an index that is not assigned to this node. This method cleans up all disk folders relating to the index
          * but does not deal with in-memory structures. For those call {@link #removeIndex}
          */
-        void deleteUnassignedIndex(String reason, IndexMetadata metadata, ClusterState clusterState);
+        void deleteUnassignedIndex(String reason, IndexMetadata metadata, @Nullable ProjectMetadata project);
```

Comment:
Can we add a javadoc for the ProjectMetadata parameter. It is not very intuitive on why it can be null.